etc/proto/pachgen/transaction_client.go

```diff
@@ -0,0 +1,108 @@
+//nolint:wrapcheck
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"path"
+	"strings"
+	"text/template"
+
+	descriptor "github.com/golang/protobuf/protoc-gen-go/descriptor"
+	plugin "github.com/golang/protobuf/protoc-gen-go/plugin"
+)
+
+func init() {
+	CodeGenerators = append(CodeGenerators, &TransactionClientGenerator{})
+}
+
+type TransactionClientGenerator struct {
+	Protos []*descriptor.FileDescriptorProto
+}
+
+var funcs = map[string]interface{}{
+	"importPath": func(proto *descriptor.FileDescriptorProto) string {
+		if *proto.Package == "taskapi" {
+			return `taskapi "github.com/pachyderm/pachyderm/v2/src/task"`
+		}
+		return fmt.Sprintf(`"github.com/pachyderm/pachyderm/v2/src/%s"`, path.Dir(*proto.Name))
+	},
+	"hasAPI": func(proto *descriptor.FileDescriptorProto) bool {
+		for _, service := range proto.Service {
+			if *service.Name == "API" || *service.Name == "Debug" {
+				return true
+			}
+		}
+		return *proto.Package == "taskapi"
```

Comment:
Need this so that we can import `taskapi`.

Reply:
seems like this is really `shouldImport`. I wonder if this would be better handled by preprocessing in the go code that collects all direct package dependencies